doc/md/writing-motoko/recursive-types.md

```diff
@@ -0,0 +1,43 @@
+---
+sidebar_position: 29
+---
+
+# Recursive types
+
+## Overview
+
+A recursive type is a type that contains the values of the same type. Recursive types enable you to create complex recursive data structures, such as linked lists or trees.
```

Comment:
Or of a type derived from it, like
``` Motoko
type Inception<T> = { #deeper : Inception<Inception<T>>; #level : Inception<T>; #shallower : T }
```
But this is currently disallowed ("is expansive") bu `moc`, so maybe my this comment is just moot :-(